SlopCrew.Server/Race/RacerManager.cs

```diff
@@ -0,0 +1,278 @@
+
+            //TODO: may be prevent double ready if it somehow happens
+            var race = GetPlayerRace(id);
+
+            if (race.State != RaceState.WaitingForPlayersToBeReady) {
+                Log.Information($"Race is not waiting for players");
+                return;
+            }
+
+            race.ConfirmedPlayers += 1;
+        }
+
+        public void AddPlayerTime(uint id, float time) {
+            if (!IsPlayerAlreadyRacing(id)) {
+                Log.Information($"Player {id} is not racing");
+                return;
+            }
+
+            var race = GetPlayerRace(id);
+
+            if (race.Ranking.ContainsKey(id)) {
+                Log.Information($"Player {id} already finished");
+                return;
+            }
+
+            race.Ranking.Add(id, time);
+        }
+
+        public bool IsPlayerAlreadyRacing(uint id) {
+            return races.Any(kv => kv.Value.Players.Any(p => p.ID == id));
+        }
+
+        /// <summary>
+        /// Useful to remove player on disconnect
+        /// </summary>
+        /// <param name="player"></param>
+        internal void RemovePlayerIfRacing(Player player) {
+            if (player == null) {
+                return;
+            }
+
+            if (!IsPlayerAlreadyRacing(player.ID)) {
+                Log.Information($"Player {player.ID} is not racing");
+                return;
+            }
+
+            var race = GetPlayerRace(player.ID);
+            race.Players.Remove(player);
+        }
+
+        public (DateTime, RaceConfig?) GetARace(Player player) {
+            if (IsPlayerAlreadyRacing(player.ID)) {
+                return (DateTime.MinValue, null);
+            }
+
+            var availableRace = races.FirstOrDefault(kv =>
+                kv.Value.State == RaceState.WaitingForPlayers
+                && kv.Value.Players.Count() < MAX_PLAYERS_IN_RACE);
+
+            if (availableRace.Key != Guid.Empty) {
+                availableRace.Value.Players.Add(player);
+                return (availableRace.Value.Initialized.AddSeconds(MAX_WAITING_JOINING_TIME_FOR_SECS), availableRace.Value.Config);
+            }
+
+            (var name, var raceConf) = GetANewRaceConf();
+
+            if (raceConf == null) {
+                return (DateTime.MinValue, null);
+            }
+
+            var newRace = new Race {
+                Config = raceConf!,
+                Players = new List<Player> { player },
+                State = RaceState.WaitingForPlayers,
+                Initialized = DateTime.UtcNow,
+                Name = name
+            };
+
+            var success = races.TryAdd(Guid.NewGuid(), newRace);
+
+            if (success) {
+                return (newRace.Initialized.AddSeconds(MAX_WAITING_JOINING_TIME_FOR_SECS), raceConf);
+            }
+
+            Log.Error("Failed to get a new race");
+
+            return (DateTime.MinValue, null);
+        }
+
+        private (string, RaceConfig?) GetANewRaceConf() {
```

Comment:
Thinking about it, a cache should be good here
No need to fetch race each time a race will start